src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmers.java

```diff
@@ -0,0 +1,135 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import scala.Tuple2;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * SparkTool to identify 63-mers in the reference that occur more than 3 times.
+ *
+ * Created by tsharpe on 12/4/15.
+ */
+@CommandLineProgramProperties(summary="Use spark to find the set of high copy number kmers in a reference.",
+        oneLineSummary="find ref kmers with high copy number",
+        programGroup = SparkProgramGroup.class)
+public final class FindBadGenomicKmers extends GATKSparkTool
+{
+    private static final long serialVersionUID = 1L;
+    private static final Long MAX_KMER_FREQ = 3L;
+    private static final int KLEN = 63;
+    private static final int REF_REC_LEN = 10000;
+    private static final long CHUNK_SIZE = 3000000;
+
+    @Argument(doc = "file for ubiquitous kmer output", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    private String outputFile;
+
+    @Override
+    public boolean requiresReference()
+    {
+        return true;
+    }
+
+    // Get the list of high copy number kmers in the reference, and write them to a file.
+    @Override
+    protected void runTool( final JavaSparkContext ctx )
+    {
+        SAMFileHeader hdr = getHeaderForReads();
```

Comment:
This tool will probably be part of a pipeline that will have reads.  If reads happen to be supplied to this tool it will allow us to check for consistent dictionaries and might catch a wrong reference error.  Or have I misunderstood how the argument to getReferenceSequenceDictionary is used?